qiskit/synthesis/clifford/clifford_decompose_layers.py

```diff
@@ -137,34 +139,36 @@ def synth_clifford_layers(
         cz_func_reverse_qubits=cz_func_reverse_qubits,
     )
 
-    layeredCircuit.append(S2_circ, qubit_list)
-
+    layered_circuit = QuantumCircuit(num_qubits)
+    layered_circuit.append(S2_circ, qubit_list, copy=False)
     if cx_cz_synth_func is None:
-        layeredCircuit.append(CZ2_circ, qubit_list)
-
+        layered_circuit.append(CZ2_circ, qubit_list, copy=False)
         CXinv = CX_circ.copy().inverse()
-        layeredCircuit.append(CXinv, qubit_list)
-
+        layered_circuit.append(CXinv, qubit_list, copy=False)
     else:
         # note that CZ2_circ is None and built into the CX_circ when
         # cx_cz_synth_func is not None
-        layeredCircuit.append(CX_circ, qubit_list)
-
-    layeredCircuit.append(H2_circ, qubit_list)
-    layeredCircuit.append(S1_circ, qubit_list)
-    layeredCircuit.append(CZ1_circ, qubit_list)
+        layered_circuit.append(CX_circ, qubit_list, copy=False)
+    layered_circuit.append(H2_circ, qubit_list, copy=False)
+    layered_circuit.append(S1_circ, qubit_list, copy=False)
+    layered_circuit.append(CZ1_circ, qubit_list, copy=False)
```

Comment:
Good idea (not copying)! I now seem to remember that either `_append` or `compose` should be even more performant (not sure).